src/react/inject.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react'
+import getWrappedDisplayName from './getWrappedDisplayName'
+
+export default injectFunction => WrappedComponent => class extends Component {
+  static displayName = `inject(${getWrappedDisplayName(WrappedComponent)}`
+
+  static contextTypes = {
+    kStore: () => null, // this is to avoid importing prop-types
+  }
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      injectedProps: {},
+    }
+  }
+
+  componentWillMount() {
+    // subscribe
+    this.unsubscribe = this.context.kStore.subscribe(() => {
+      this.inject()
+    })
+
+    // run in once
+    this.inject()
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.readStore(nextProps)()
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe()
+  }
+
+  inject = () => {
+    this.setState(state => ({
```

Comment:
He I'd like to call `shallowEquals` to test both `this.state.injectedProps` and the new value returned by `injectFunction` before calling `this.setState`, as an optimisation.